assignment4/disjointset.py

```diff
@@ -0,0 +1,37 @@
+class DisjointSet:
+    """Implementation of a Disjoint Set. Takes integers as keys."""
+
+    def __init__(self):
+        self.parent = list()
+        self.rank = list()
+        self.setAmount = 0
+
+    def getSetAmount(self):
+        return self.setAmount
+
+    def makeSet(self, v):
+        if len(self.parent) <= v:
```

Comment:
It might make it more readable if you had a helper method for this.
For example:
```python
  def makeSet(self, v):
    self._ensureAtLeastSize(v)
    self.parent[v] = v
    self.rank[v] = 0
    self.setAmount = 1
```